In [281]:
import pandas as pd
import os
import numpy as np
from pprint import pprint

# ajustando algumas configurações para prints
pd.set_option('display.max_columns', None)        # todas as colunas
pd.set_option('display.max_rows', None)           # todas as linhas
pd.set_option('display.width', None)              # largura automática
pd.set_option('display.max_colwidth', None)       # conteúdo completo das células
pd.set_option('display.expand_frame_repr', False) # evita quebrar dataframe em várias linhas


In [282]:
# pegando a pasta de trabalho para poder extrair os dados do dataset
PASTA_TRABALHO = os.path.dirname(os.path.abspath(os.getcwd()))

## Lendo e armazenando todas as tabelas dos anos necessários

In [283]:
# lista com os nomes das pastas que contem os datasets
nomes_arquivos = ["2024", "2025", "2026"]

# dicionario que contêm todos os dataframes do respectivo ano no seguinte formato:
"""
    ano1:
        ocorrencia_ano1,
        acorrencia_pessoa_ano1,
        acorrencia_todas_causas_ano1
    ano2:
        ...
"""
dataframes_por_ano = {}

# percorre e salva todos as dataframes dos anos escolhidos
for ano in nomes_arquivos:
    diretorio_dataframe = os.path.join(PASTA_TRABALHO, "dataset", ano)

    # percorre todas as tabelas que existem nesse ano (só existe um arquivo)
    for tabela in os.listdir(diretorio_dataframe):

        caminho = os.path.join(diretorio_dataframe, tabela)
        data_frame = pd.read_csv(caminho, sep=";", encoding="latin1") # adiciona a tabela em questão ao dicionário temporário

    dataframes_por_ano[ano] = data_frame # adiciona a tabela de dados ao respectivo ano


## juntando as tabelas distintas dos 3 anos em somente uma tabela

In [284]:
dataset = pd.concat([df for df in dataframes_por_ano.values()])

## Alterando os tipos de dados para não ter nunhum tipo de conflito com as transformações e armazenar de forma correta os dados

In [ ]:
print(dataset["km"].map(type).value_counts())

# DATA
dataset["data_inversa"] = pd.to_datetime(
    dataset["data_inversa"],
    format="%Y-%m-%d"
)


# HORÁRIO
dataset["horario"] = pd.to_datetime(
    dataset["horario"],
    format="%H:%M:%S"
).dt.time


# VALORES NUMÉRICOS
# trocando a virgula por ponto para poder converter para float
dataset["latitude"] = pd.to_numeric(
    dataset["latitude"].astype(str).str.replace(",", "."),
    errors="coerce"
)

dataset["longitude"] = pd.to_numeric(
    dataset["longitude"].astype(str).str.replace(",", "."),
    errors="coerce"
)

# passando para o tipo numérico
dataset["latitude"] = pd.to_numeric(
    dataset["latitude"],
    errors="coerce"
)

dataset["longitude"] = pd.to_numeric(
    dataset["longitude"],
    errors="coerce"
)

dataset["km"] = pd.to_numeric(
    dataset["km"].astype(str).str.replace(",", "."),
    errors="coerce"
)

dataset["km"] = pd.to_numeric(
    dataset["km"],
    errors="coerce"
)

km
<class 'str'>    455060
Name: count, dtype: int64


km
0.0       2157
1.0       1929
2.0       1877
3.0       1823
5.0       1657
4.0       1646
7.0       1263
6.0       1209
19.0      1058
10.0      1037
16.0      1036
22.0      1031
17.0      1013
8.0       1008
12.0      1005
11.0       993
13.0       992
119.0      986
9.0        960
117.0      959
15.0       958
115.0      957
14.0       951
21.0       934
18.0       917
20.0       917
70.0       883
58.0       874
121.0      861
83.0       857
123.0      852
65.0       850
122.0      849
145.0      845
147.0      827
118.0      820
104.0      820
120.0      817
82.0       810
124.0      803
69.0       801
25.0       798
146.0      797
112.0      797
116.0      786
148.0      784
144.0      783
105.0      783
94.0       782
95.0       781
50.0       781
23.0       781
24.0       778
51.0       771
86.0       771
109.0      769
64.0       768
108.0      762
62.0       751
110.0      732
88.0       722
46.0       719
92.0       717
67.0       716
80.0       708
61.0       708
138.0  

In [286]:
# printando algumas informações pertinentes que ajudam a avaliar quais transformações são necessárias
display(dataset.shape)
display(dataset.head())
display(dataset.info())
display(dataset.describe())
display(dataset.isnull().sum())

(455060, 35)

,id,pesid,data_inversa,dia_semana,horario,uf,br,km,municipio,causa_acidente,tipo_acidente,classificacao_acidente,fase_dia,sentido_via,condicao_metereologica,tipo_pista,tracado_via,uso_solo,id_veiculo,tipo_veiculo,marca,ano_fabricacao_veiculo,tipo_envolvido,estado_fisico,idade,sexo,ilesos,feridos_leves,feridos_graves,mortos,latitude,longitude,regional,delegacia,uop
0,571789.0,1269011,2024-01-01,segunda-feira,03:56:00,ES,101,38.0,CONCEICAO DA BARRA,Ultrapassagem Indevida,Colisão lateral sentido oposto,NaN,Plena Noite,Crescente,Céu Claro,Simples,Reta,Não,1018240,Caminhão-trator,M.BENZ/AXOR 2544 LS,2014,Condutor,Ileso,46,Masculino,1,0,0,0,-18.482610,-39.923790,SPRF-ES,DEL04-ES,UOP02-DEL04-ES
1,571804.0,1269118,2024-01-01,segunda-feira,04:50:00,PI,343,185.0,PIRIPIRI,Manobra de mudança de faixa,Colisão frontal,Com Vítimas Fatais,Amanhecer,Decrescente,Céu Claro,Simples,Reta,Sim,1018325,Caminhão,M.BENZ/ATEGO 2426,2019,Condutor,Ileso,24,Masculino,1,0,0,0,-4.296033,-41.767327,SPRF-PI,DEL02-PI,UOP01-DEL02-PI
2,571806.0,1269735,2024-01-01,segunda-feira,04:30:00,BA,116,578.0,BREJOES,Ingestão de álcool pelo condutor,Colisão frontal,Com Vítimas Fatais,Plena Noite,Decrescente,Céu Claro,Simples,Curva,Não,1018705,Caminhão-trator,SCANIA/R450 A6X2,2021,Condutor,Ileso,45,Masculino,1,0,0,0,-13.071583,-39.961111,SPRF-BA,DEL03-BA,UOP02-DEL03-BA
3,571818.0,1269075,2024-01-01,segunda-feira,06:30:00,SE,101,18.0,MALHADA DOS BOIS,Reação tardia ou ineficiente do condutor,Saída de leito carroçável,Com Vítimas Feridas,Amanhecer,Crescente,Céu Claro,Dupla,Declive;Reta,Não,1018287,Caminhão-trator,IVECO/STRALIS 800S48TZ,2022,Condutor,Lesões Graves,33,Masculino,0,0,1,0,-10.356019,-36.905522,SPRF-SE,DEL02-SE,UOP02-DEL02-SE
4,571838.0,1269159,2024-01-01,segunda-feira,05:00:00,MT,364,240.0,RONDONOPOLIS,Condutor deixou de manter distância do veículo da frente,Colisão traseira,Sem Vítimas,Pleno dia,Crescente,Céu Claro,Dupla,Reta,Não,1018348,Caminhão,SCANIA/P 310 B8X2,2013,Condutor,Ileso,57,Masculino,1,0,0,0,-16.179141,-54.789053,SPRF-MT,DEL02-MT,UOP01-DEL02-MT


<class 'pandas.core.frame.DataFrame'>
Index: 455060 entries, 0 to 64124
Data columns (total 35 columns):
 #   Column                  Non-Null Count   Dtype         
---  ------                  --------------   -----         
 0   id                      455060 non-null  float64       
 1   pesid                   455060 non-null  int64         
 2   data_inversa            455060 non-null  datetime64[ns]
 3   dia_semana              455060 non-null  object        
 4   horario                 455060 non-null  object        
 5   uf                      455060 non-null  object        
 6   br                      455060 non-null  int64         
 7   km                      455060 non-null  float64       
 8   municipio               455060 non-null  object        
 9   causa_acidente          455060 non-null  object        
 10  tipo_acidente           455060 non-null  object        
 11  classificacao_acidente  455045 non-null  object        
 12  fase_dia                455060 non-n

None

,id,pesid,data_inversa,br,km,id_veiculo,ano_fabricacao_veiculo,idade,ilesos,feridos_leves,feridos_graves,mortos,latitude,longitude
count,455060.00000,4.550600e+05,455060,455060.000000,455060.000000,4.550600e+05,455060.000000,455060.000000,455060.000000,455060.000000,455060.000000,455060.000000,455060.000000,455060.000000
mean,670834.11838,1.369358e+06,2025-03-04 05:03:41.161165056,211.183125,268.202186,1.160016e+06,1834.960025,33.417624,0.392267,0.327104,0.102802,0.031031,-18.648385,-46.631066
min,571772.00000,0.000000e+00,2024-01-01 00:00:00,0.000000,0.000000,0.000000e+00,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,-33.689326,-72.835751
25%,619713.00000,1.354106e+06,2024-08-04 00:00:00,101.000000,81.000000,1.101625e+06,2007.000000,21.000000,0.000000,0.000000,0.000000,0.000000,-24.982890,-50.576593
50%,665497.00000,1.479008e+06,2025-03-03 00:00:00,163.000000,203.000000,1.191750e+06,2013.000000,34.000000,0.000000,0.000000,0.000000,0.000000,-20.225260,-47.481389
75%,723575.00000,1.604566e+06,2025-10-05 00:00:00,324.000000,421.000000,1.286620e+06,2020.000000,47.000000,1.000000,1.000000,0.000000,0.000000,-12.540656,-42.466417
max,772176.00000,1.737879e+06,2026-04-30 00:00:00,495.000000,1470.000000,1.383658e+06,2026.000000,2025.000000,1.000000,1.000000,1.000000,1.000000,4.461419,-32.406822
std,58377.94562,4.431653e+05,NaN,127.920838,231.231042,2.360007e+05,573.047374,41.071743,0.488256,0.469156,0.303700,0.173402,7.679181,6.246490


id                            0
pesid                         0
data_inversa                  0
dia_semana                    0
horario                       0
uf                            0
br                            0
km                            0
municipio                     0
causa_acidente                0
tipo_acidente                 0
classificacao_acidente       15
fase_dia                      0
sentido_via                   0
condicao_metereologica        0
tipo_pista                    0
tracado_via                   0
uso_solo                      0
id_veiculo                    0
tipo_veiculo              14840
marca                         0
ano_fabricacao_veiculo        0
tipo_envolvido            39942
estado_fisico             39942
idade                         0
sexo                      39942
ilesos                        0
feridos_leves                 0
feridos_graves                0
mortos                        0
latitude                      0
longitud

## Transformando a coluna ano veículo
Olhando para as informações do discribe() é possível observar que o valor mínimo é 0, ou seja veiculos sem a informação de ano, no lugar de 0 deveria existir null para não influenciar calculos de métricas posteriormente e saber da ausência de informação

In [287]:
dataset.loc[dataset["ano_fabricacao_veiculo"] == 0, "ano_fabricacao_veiculo"] = pd.NA

## Avaliando booleanos de condição física das pessoas envolvidas
Existem 4 colunas booleanas que representam a condição física da pessoa envolvida, entretanto essa mesma informação está na coluna `estado_fisico` de forma textual

In [288]:
pd.crosstab( # para verificar se realmente as informaçõs de uma coluna estão em outra
    dataset["estado_fisico"],
    [
        dataset["ilesos"],
        dataset["feridos_leves"],
        dataset["feridos_graves"],
        dataset["mortos"]
    ]
)

dataset = dataset.drop(columns=[
    "ilesos",
    "feridos_leves",
    "feridos_graves",
    "mortos"
])

## Avaliando os valores da coluna idade

Nessa coluna tem dois problemas principais:
- Quando não existe informação de qual a idade ela é preenchida como -1, é melhor fica como NULL pois deixar esse campo com -1 pode atrapalhar o calculo de métricas posteriormente como saber qual a média de idade das pessoas que sofrem acidentes
- Existem idades que são impossíveis como por exemplo maiores que 200 anos, avaliando os dados notem que existem amostras com 127 anos depois com 225 anos, apesar de idades superiores a 110 anos já serem extremamente improváveis e a pessoa mais velha registrada ter 122 anos vou considerar o limite como 127
- um ponto que provavelmente deve ser um erro, mas não tem como saber de forma 100% correta são as pessoas com 0 anos, isso poderia indicar pessoas com menos de 1 ano de idade, entretando como esse valor detem mais de 78.000 amostras enquanto a quantidade e pessoas maiores que 0 e menor ou igual a 25 anos é de aproximadamente 73.000 leva a indicar que provavelmente esses dados foram preenchidos de forma errada

In [289]:
# analisando os valores distintos dessa coluna para saber quando existem prováveis outliers
# display(dataset["idade"].value_counts().sort_index())

# analisando a quantidade de pessoas com idade igual a 0 e comparando com outro grupo
idade_0_a_25 = dataset.loc[(dataset["idade"] <= 25) & (dataset["idade"] > 0), "idade" ].count()
idade_0 = dataset.loc[dataset["idade"] == 0, "idade"].count()
print(f"Quantidade de pessoas com 0 (zero) anos de idade: {idade_0}")
print(f"Quantidade de pessoas com idades entre 1-25 anos: {idade_0_a_25}")
# apesar de ser inprovável que a quantidade de vítimas recem nascidas seja superior as de idade entre 1-25 anos não é possível dizer que isso é um erro de forma categorica

# substituindo os valores -1 por NULL
dataset.loc[dataset["idade"] == -1, "idade"] = pd.NA

# substituindo idade impossíveis
dataset.loc[dataset["idade"] > 127, "idade"] = pd.NA

Quantidade de pessoas com 0 (zero) anos de idade: 78970
Quantidade de pessoas com idades entre 1-25 anos: 73925


# Tratando a coluna classificacao_acidente
Existem alguns valores nulos nessa coluna que possivelmente podem ser passados para um tipo dentro das categorias já existentes

In [290]:
# De acordo com o dicionário de dados existe a classificação "Ignorado" para essa coluna, vamos passar os valores NULL para essa classe
dataset["classificacao_acidente"] = dataset["classificacao_acidente"].fillna("Ignorado")

# Analisando a coluna "tracado_via"
Analisar se essa coluna tem multiplos valores em uma única célula e quais são os valores possíveis

In [291]:
display(dataset["tracado_via"].unique())
"""
provavelmente vou ter que tratar essa coluna descobrindo quais são os valores atómicos usados, criar uma tabela com cada um dos tipos de traçado contendo IDs para eles e criar uma tabela ponte que
vai ligar o acidente as suas causas
"""

array(['Reta', 'Curva', 'Declive;Reta', 'Aclive;Curva', 'Reta;Declive',
       'Retorno Regulamentado', 'Curva;Declive', 'Declive;Curva',
       'Reta;Ponte', 'Aclive', 'Reta;Aclive',
       'Retorno Regulamentado;Reta', 'Viaduto', 'Curva;Viaduto',
       'Reta;Retorno Regulamentado', 'Interseção de Vias;Reta',
       'Aclive;Reta', 'Reta;Ponte;Declive',
       'Interseção de Vias;Rotatória', 'Rotatória;Reta', 'Ponte',
       'Rotatória', 'Declive', 'Interseção de Vias',
       'Curva;Em Obras;Desvio Temporário', 'Declive;Reta;Ponte',
       'Reta;Interseção de Vias;Em Obras', 'Aclive;Retorno Regulamentado',
       'Retorno Regulamentado;Reta;Interseção de Vias', 'Curva;Aclive',
       'Reta;Em Obras;Desvio Temporário', 'Curva;Aclive;Em Obras',
       'Curva;Em Obras', 'Reta;Em Obras;Ponte', 'Reta;Rotatória',
       'Curva;Interseção de Vias', 'Reta;Viaduto',
       'Declive;Interseção de Vias', 'Reta;Interseção de Vias',
       'Aclive;Rotatória;Curva', 'Aclive;Ponte;Reta',
       'Re

'\nprovavelmente vou ter que tratar essa coluna descobrindo quais são os valores atómicos usados, criar uma tabela com cada um dos tipos de traçado contendo IDs para eles e criar uma tabela ponte que\nvai ligar o acidente as suas causas\n'

## Tratando os valores nulos das colunas: tipo_envolvido, stado_fisico e sexo
Como cada uma dessas colunas contem a mesma quantidade de valores nulos pode ser que sejam exatamente a mesma linha que não contem multiplas informações, uma pessoa que não contem nenhuma dessas informações pode ser caracterizada como não identificada para facilitar o entendimento das informações

In [292]:
# verificando se realmente é a mesma linha com multiplas linhas colunas vazia
filtro = (
    dataset["estado_fisico"].isna() &
    dataset["tipo_envolvido"].isna() &
    dataset["sexo"].isna()
)
print(f"quantidade de linhas que não tem informação nas colunas em questão: {filtro.sum()}")

# agora que foi mostrado que são a mesma linha vou avaliar qual a idade descrita, levando em consideração que essa é a ultima informação que sobre referente a pessoa envolvida
print(f"desas linhas quais as idades das pessoas em questão: {dataset.loc[filtro, "idade"].unique()}")

# avaliando quantos pesid distintos existem para essa pessoa
print(f"quantidade de pesid existentes para cada uma dessas linhas: {dataset.loc[filtro, "pesid"].nunique()}")

# avaliando quantos id_veiculo distintos existem para essa pessoa
print(f"quantidade de id_veiculo existentes para cada uma dessas linhas: {dataset.loc[filtro, "id_veiculo"].nunique()}")

# avaliando o tipo do veiculo envolvido pera essas pessoas
print(f"quantidade de veiculos sem tipo definido para essas pessoas: {dataset.loc[filtro, "tipo_veiculo"].isna().sum()}")

# aparentemente quando uma pessoa não é identificada ela é colocada como um ID específico e com os valores nulos nessas respectivas colunas e idade igual a 0 (zero)
# criando uma coluna nova que vai dizer se a pessoa em questão foi ou não identificada para facilitar o entendimento dos dados
dataset["identificada"] = 1
dataset.loc[filtro, "identificada"] = 0

quantidade de linhas que não tem informação nas colunas em questão: 39942
desas linhas quais as idades das pessoas em questão: [0.]
quantidade de pesid existentes para cada uma dessas linhas: 1
quantidade de id_veiculo existentes para cada uma dessas linhas: 39942
quantidade de veiculos sem tipo definido para essas pessoas: 0


In [293]:
# printando algumas informações pertinentes que ajudam a avaliar quais transformações são necessárias
display(dataset.shape)
display(dataset.head())
display(dataset.info())
display(dataset.describe())
display(dataset.isnull().sum())

(455060, 32)

,id,pesid,data_inversa,dia_semana,horario,uf,br,km,municipio,causa_acidente,tipo_acidente,classificacao_acidente,fase_dia,sentido_via,condicao_metereologica,tipo_pista,tracado_via,uso_solo,id_veiculo,tipo_veiculo,marca,ano_fabricacao_veiculo,tipo_envolvido,estado_fisico,idade,sexo,latitude,longitude,regional,delegacia,uop,identificada
0,571789.0,1269011,2024-01-01,segunda-feira,03:56:00,ES,101,38.0,CONCEICAO DA BARRA,Ultrapassagem Indevida,Colisão lateral sentido oposto,Ignorado,Plena Noite,Crescente,Céu Claro,Simples,Reta,Não,1018240,Caminhão-trator,M.BENZ/AXOR 2544 LS,2014.0,Condutor,Ileso,46.0,Masculino,-18.482610,-39.923790,SPRF-ES,DEL04-ES,UOP02-DEL04-ES,1
1,571804.0,1269118,2024-01-01,segunda-feira,04:50:00,PI,343,185.0,PIRIPIRI,Manobra de mudança de faixa,Colisão frontal,Com Vítimas Fatais,Amanhecer,Decrescente,Céu Claro,Simples,Reta,Sim,1018325,Caminhão,M.BENZ/ATEGO 2426,2019.0,Condutor,Ileso,24.0,Masculino,-4.296033,-41.767327,SPRF-PI,DEL02-PI,UOP01-DEL02-PI,1
2,571806.0,1269735,2024-01-01,segunda-feira,04:30:00,BA,116,578.0,BREJOES,Ingestão de álcool pelo condutor,Colisão frontal,Com Vítimas Fatais,Plena Noite,Decrescente,Céu Claro,Simples,Curva,Não,1018705,Caminhão-trator,SCANIA/R450 A6X2,2021.0,Condutor,Ileso,45.0,Masculino,-13.071583,-39.961111,SPRF-BA,DEL03-BA,UOP02-DEL03-BA,1
3,571818.0,1269075,2024-01-01,segunda-feira,06:30:00,SE,101,18.0,MALHADA DOS BOIS,Reação tardia ou ineficiente do condutor,Saída de leito carroçável,Com Vítimas Feridas,Amanhecer,Crescente,Céu Claro,Dupla,Declive;Reta,Não,1018287,Caminhão-trator,IVECO/STRALIS 800S48TZ,2022.0,Condutor,Lesões Graves,33.0,Masculino,-10.356019,-36.905522,SPRF-SE,DEL02-SE,UOP02-DEL02-SE,1
4,571838.0,1269159,2024-01-01,segunda-feira,05:00:00,MT,364,240.0,RONDONOPOLIS,Condutor deixou de manter distância do veículo da frente,Colisão traseira,Sem Vítimas,Pleno dia,Crescente,Céu Claro,Dupla,Reta,Não,1018348,Caminhão,SCANIA/P 310 B8X2,2013.0,Condutor,Ileso,57.0,Masculino,-16.179141,-54.789053,SPRF-MT,DEL02-MT,UOP01-DEL02-MT,1


<class 'pandas.core.frame.DataFrame'>
Index: 455060 entries, 0 to 64124
Data columns (total 32 columns):
 #   Column                  Non-Null Count   Dtype         
---  ------                  --------------   -----         
 0   id                      455060 non-null  float64       
 1   pesid                   455060 non-null  int64         
 2   data_inversa            455060 non-null  datetime64[ns]
 3   dia_semana              455060 non-null  object        
 4   horario                 455060 non-null  object        
 5   uf                      455060 non-null  object        
 6   br                      455060 non-null  int64         
 7   km                      455060 non-null  float64       
 8   municipio               455060 non-null  object        
 9   causa_acidente          455060 non-null  object        
 10  tipo_acidente           455060 non-null  object        
 11  classificacao_acidente  455060 non-null  object        
 12  fase_dia                455060 non-n

None

,id,pesid,data_inversa,br,km,id_veiculo,ano_fabricacao_veiculo,idade,latitude,longitude,identificada
count,455060.00000,4.550600e+05,455060,455060.000000,455060.000000,4.550600e+05,414630.000000,454742.000000,455060.000000,455060.000000,455060.000000
mean,670834.11838,1.369358e+06,2025-03-04 05:03:41.161165056,211.183125,268.202186,1.160016e+06,2013.884449,32.557024,-18.648385,-46.631066,0.912227
min,571772.00000,0.000000e+00,2024-01-01 00:00:00,0.000000,0.000000,0.000000e+00,1900.000000,0.000000,-33.689326,-72.835751,0.000000
25%,619713.00000,1.354106e+06,2024-08-04 00:00:00,101.000000,81.000000,1.101625e+06,2009.000000,21.000000,-24.982890,-50.576593,1.000000
50%,665497.00000,1.479008e+06,2025-03-03 00:00:00,163.000000,203.000000,1.191750e+06,2014.000000,34.000000,-20.225260,-47.481389,1.000000
75%,723575.00000,1.604566e+06,2025-10-05 00:00:00,324.000000,421.000000,1.286620e+06,2021.000000,47.000000,-12.540656,-42.466417,1.000000
max,772176.00000,1.737879e+06,2026-04-30 00:00:00,495.000000,1470.000000,1.383658e+06,2026.000000,127.000000,4.461419,-32.406822,1.000000
std,58377.94562,4.431653e+05,NaN,127.920838,231.231042,2.360007e+05,8.341989,20.563964,7.679181,6.246490,0.282965


id                            0
pesid                         0
data_inversa                  0
dia_semana                    0
horario                       0
uf                            0
br                            0
km                            0
municipio                     0
causa_acidente                0
tipo_acidente                 0
classificacao_acidente        0
fase_dia                      0
sentido_via                   0
condicao_metereologica        0
tipo_pista                    0
tracado_via                   0
uso_solo                      0
id_veiculo                    0
tipo_veiculo              14840
marca                         0
ano_fabricacao_veiculo    40430
tipo_envolvido            39942
estado_fisico             39942
idade                       318
sexo                      39942
latitude                      0
longitude                     0
regional                     23
delegacia                   168
uop                         536
identifi